common/ferrostar/src/routing_adapters/osrm/mod.rs

```diff
@@ -81,6 +81,14 @@ impl Route {
             })
             .collect();
 
+        Self::from_osrm_route(route, &waypoints, polyline_precision)
+    }
+
+    pub fn from_osrm_route(
+        route: &OsrmRoute,
+        waypoints: &Vec<Waypoint>,
```

Comment:
I think we can change this like so to make it slightly more generic (will work for any slice type, not exclusively vectors by reference).

```suggestion
        waypoints: &[Waypoint],
```